### ISSUE TITLE
"Waiting for SSH connection (slow)..." handler fails when provisioning staging

### ISSUE THREAD

**Issue by garrettr** [freedomofpress/securedrop#1977]
# Bug

## Description

This just started occurring this afternoon, so it was probably introduced by #1974.

## Steps to Reproduce

1. Do a clean provision of the staging environment: `vagrant destroy -f /staging/ && vagrant up /staging/`.
2. Wait until the playbook reaches the `tor-hidden-services` role.

## Expected Behavior

The `tor-hidden-services` role completes all of its tasks successfully and playbook execution continues to the next role.

## Actual Behavior

The `Waiting for SSH connection (slow)...` handler hangs for a long time, and eventually fails with the following message:

```
RUNNING HANDLER [tor-hidden-services : Waiting for SSH connection (slow)...] ***
fatal: [app-staging -> localhost]: FAILED! => {"changed": false, "elapsed": 300, "failed": true, "msg": "Timeout when waiting for search string OpenSSH in 127.0.0.1:22"}
fatal: [mon-staging -> localhost]: FAILED! => {"changed": false, "elapsed": 300, "failed": true, "msg": "Timeout when waiting for search string OpenSSH in 127.0.0.1:22"}
```

**Comment by conorsch** [comment]
Came in via 2416995d9 in #1974. The problem is the disparity between two nearly-equivalent options:

1. `ansible_ssh_port`
2. `ansible_port`

Ansible v2 introduced the preferred 2, deprecating 1 in the process. Vagrant, however, still hardcodes `ansible_ssh_port` in its automatically generated INI-style inventory file used by Ansible. Since CI passed without issue on #1974, seems like the CI tooling is using the newer port declaration, while local VMs in Vagrant still use the older.

We should support both by chaining `|default()` filters.

**Comment by garrettr** [comment]
> We should support both by chaining |default() filters.

It looks like that's what the code [did](https://github.com/freedomofpress/securedrop/commit/2416995d9c05b90295e7a3e50f080b9409c4b5c8#diff-59f750f71db04261d72926b287cf19f9L46) prior to 2416995d9c05b90295e7a3e50f080b9409c4b5c8. @msheiny Do you remember why you had to make that change?

**Comment by conorsch** [comment]
Looks like premature optimization to me, judging by the commit message. @msheiny is right, the `ansible_ssh_port` _is_ deprecated, but we need to honor it due to Vagrant dragging its feet on the transition. For the sake of consistency, I retained usage of `ansible_ssh_port` in the [new dynamic inventory file](https://github.com/freedomofpress/securedrop/pull/1779/files#diff-4993403b5ac197d2b65445d49178b22bR114), so we definitely need to honor it.

**Comment by msheiny** [comment]
Yeeepppp -- it was oversight on my part regarding vagrant's usage of deprecated vars :(

**Comment by conorsch** [comment]
Resolved via #1980.
